### PR TITLE
add: java hashCode BKDR

### PR DESCRIPTION
**Pull request by Penglq**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hash/base_hash_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+func TestBKDRHash(t *testing.T) {
+	valTable := []struct {
+		in   []byte
+		want int32
+	}{
+		{[]byte(""), 0},
+		{[]byte("ab"), 3105},
+		{[]byte("12345678"), -1861353340},
+		{[]byte("ZD20221202XFSDSA"), -1214774504},
+		{[]byte("www.eeee.com/"), -1766224091},
```
Comment (jjeffcaii) [inline hash/base_hash_test.go:37]: It would be better if there was a test case for Chinese chars, for example: `Hello, 世界`.

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Review by cjphaha — APPROVED** [review]

File: hash/base_hash_test.go (hunk 1 above)
Comment (Penglq) [reply hash/base_hash_test.go:37]: ok, I will add it.

Comment (Penglq) [reply hash/base_hash_test.go]: It should use rune.

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**